docs/cmdline-opts/ipfs-gateway.d

```diff
@@ -0,0 +1,26 @@
+c: Copyright (C) 2022, Mark Gaiser, <mark.gaiser@protocol.ai>
+SPDX-License-Identifier: curl
+Long: ipfs-gateway
+Arg: <URL>
+Help: Defines the gateway to handle IPFS protocol requests.
+Added: 7.87.0
+See-also: help manual
+Category: ipfs
+Example: --ipfs-gateway $URL ipfs://
+---
+Specifies which gateway to use for resolving ipfs:// and ipns://.
+Not specifying this argument will let cURL try to automatically
+find a local gateway in your homefolder in ~/.ipfs/gateway.
+
+If you run a local IPFS node, this gateway is by default
+available under http://localhost:8080. A full example URL would
+look like:
+curl --ipfs-gateway http://localhost:8080 ipfs://bafybeigagd5nmnn2iys2f3doro7ydrevyr2mzarwidgadawmamiteydbzi
+
+You can also specify publicly available gateways. One such
+gateway is https://ipfs.io. A full example url would look like:
+curl --ipfs-gateway https://ipfs.io ipfs://bafybeigagd5nmnn2iys2f3doro7ydrevyr2mzarwidgadawmamiteydbzi
+
+There are many public IPFS gateways. As a starting point to find
+one that works for your case, consult this page:
+https://ipfs.github.io/public-gateway-checker/
```

Comment:
I would also like a caution added that mentions that gateways can inspect and interfere with traffic without curl noticing. I don't think users will be aware otherwise.

Reply:
Very good point! Fixed in next push (also refers to https://docs.ipfs.tech/reference/http/gateway/#trusted-vs-trustless)